core/src/main/java/org/apache/iceberg/jdbc/JdbcTableOperations.java

```diff
@@ -0,0 +1,220 @@
+    } catch (SQLTransientConnectionException | SQLNonTransientConnectionException e) {
+      throw new UncheckedSQLException(e, "Database Connection failed");
+    } catch (DataTruncation e) {
+      throw new UncheckedSQLException(e, "Database data truncation error");
+    } catch (SQLWarning e) {
+      throw new UncheckedSQLException(e, "Database warning");
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Failed to connect to database");
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e, "Interrupted during commit");
+    }
+  }
+
+  private void updateTable(String newMetadataLocation, String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    int updatedRecords = connections.run(conn -> {
+      try (PreparedStatement sql = conn.prepareStatement(JdbcUtil.DO_COMMIT_SQL)) {
+        // UPDATE
+        sql.setString(1, newMetadataLocation);
+        sql.setString(2, oldMetadataLocation);
+        // WHERE
+        sql.setString(3, catalogName);
+        sql.setString(4, JdbcUtil.namespaceToString(tableIdentifier.namespace()));
+        sql.setString(5, tableIdentifier.name());
+        sql.setString(6, oldMetadataLocation);
+        return sql.executeUpdate();
+      }
+    });
+
+    if (updatedRecords == 1) {
+      LOG.debug("Successfully committed to existing table: {}", tableIdentifier);
+    } else {
+      throw new CommitFailedException("Failed to update the table %s from catalog %s " +
+          "Maybe another process changed it", tableIdentifier, catalogName);
+    }
+
+  }
+
+  private void createTable(String newMetadataLocation) throws SQLException, InterruptedException {
+    int insertRecord = connections.run(conn -> {
+      try (PreparedStatement sql = conn.prepareStatement(JdbcUtil.DO_COMMIT_CREATE_TABLE_SQL)) {
+        sql.setString(1, catalogName);
+        sql.setString(2, JdbcUtil.namespaceToString(tableIdentifier.namespace()));
+        sql.setString(3, tableIdentifier.name());
+        sql.setString(4, newMetadataLocation);
+        return sql.executeUpdate();
+      }
+    });
+
+    if (insertRecord == 1) {
+      LOG.debug("Successfully committed to new table: {}", tableIdentifier);
+    } else {
+      throw new CommitFailedException("Failed to create table %s catalog %s", tableIdentifier, catalogName);
+    }
+  }
+
+  private void validateMetadataLocation(Map<String, String> table, TableMetadata base) {
+    String catalogMetadataLocation = !table.isEmpty() ? table.get(JdbcUtil.METADATA_LOCATION) : null;
+    String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+
+    if (!Objects.equals(baseMetadataLocation, catalogMetadataLocation)) {
+      throw new CommitFailedException(
+          "Cannot commit %s because base metadata location '%s' is not same as the current Catalog location '%s'",
```

Comment:
Nit: error message could be more concise: `"Cannot commit %s: metadata location %s has changed from %s"`

Reply:
updated